libmikmod/playercode/mplayer.c

```diff
@@ -3015,13 +3120,18 @@ static void pt_Notes(MODULE *mod)
 				break;
 			case UNI_INSTRUMENT:
 				inst=UniGetByte();
-				if (inst>=mod->numins) break; /* safety valve */
+				if (inst>=mod->numins) {
+					a->main.i=NULL;
+					a->main.s=NULL;
+					a->main.sample=-1;
+					break; /* safety valve */
+				}
```

Comment:
I don't have any useful examples right now but I *think* this might be completely format-dependent. Judging just on how libxmp does this (not necessarily correct, but generally pretty good):

* MOD: completely cuts the playing note.
* XM/other XM-likes: sets the volume to 0 but continues playing the current note.
* S3M/other S3M-likes: ignores the invalid instrument, continues playing the current note.
* IT (sample mode): sets the volume to 0 but continues playing the current note.
* IT (instrument mode): ignores the invalid instrument, continues playing the current note.
* MED: completely cuts the playing note.

If the above behavior descriptions are accurate, this change *could* be at the expense of S3M, XM, and IT accuracy.